Documentation/Getting-Started/glossary.md

```diff
@@ -0,0 +1,120 @@
+
+Rook allows creation and customization of storage clusters through the custom resource definitions (CRDs). For more information about the different modes of creation of clusters, refer our [documentation](https://rook.io/docs/rook/latest/CRDs/Cluster/ceph-cluster-crd/).
+
+### Host Storage Cluster
+
+A [host storage cluster](https://rook.io/docs/rook/latest/CRDs/Cluster/host-cluster/) is one where Rook configures Ceph to store data directly on the host.
+
+### PVC Storage Cluster
+
+In a [PVC-based cluster](https://rook.io/docs/rook/latest/CRDs/Cluster/pvc-cluster/), the Ceph persistent data is stored on volumes requested from a storage class of your choice.
+
+### Stretch Storage Cluster
+
+For environments that only have two failure domains available where data can be replicated, consider the case where one failure domain is down and the data is still fully available in the remaining failure domain. To support this scenario, Ceph has integrated support for [stretch clusters](https://rook.io/docs/rook/latest/CRDs/Cluster/stretch-cluster/).
+
+### External Storage Cluster
+
+An [external cluster](https://rook.io/docs/rook/latest/CRDs/Cluster/external-cluster/) is a Ceph configuration that is managed outside of the local K8s cluster.
+
+### CephBlockPool CRD
+
+Rook allows creation and customization of storage pools through the custom resource definitions (CRDs). The following settings are available for pools. For more information and examples refer this [documentation](https://rook.io/docs/rook/latest/CRDs/Block-Storage/ceph-block-pool-crd/).
+
+### CephRBDMirror CRD
+
+Rook allows creation and updating rbd-mirror daemon(s) through the custom resource definitions (CRDs). Here's an [example](https://rook.io/docs/rook/latest/CRDs/Block-Storage/ceph-rbd-mirror-crd/#creating-daemons) for the same.
+
+### CephFilesystem CRD
+
+Rook allows creation and customization of shared filesystems through the custom resource definitions (CRDs). Check out the examples [here](https://rook.io/docs/rook/latest/CRDs/Shared-Filesystem/ceph-filesystem-crd/#examples).
+
+### CephFilesystemMirror CRD
+
+Rook allows [creation](https://rook.io/docs/rook/latest/CRDs/Shared-Filesystem/ceph-fs-mirror-crd/#creating-daemon) and updating the Ceph fs-mirror daemon through the custom resource definitions (CRDs).
+
+### CephObjectStore CRD
+
+Rook allows [creation](https://rook.io/docs/rook/latest/CRDs/Object-Storage/ceph-object-store-crd/#example) and customization of object stores through the custom resource definitions (CRDs).
+
+### CephClient CRD
+
+Rook allows [creation](https://rook.io/docs/rook/latest/CRDs/ceph-client-crd/) and updating clients through the custom resource definitions (CRDs).
+
+### CephNFS CRD
+
+Rook allows exporting NFS shares of a CephFilesystem or CephObjectStore through the CephNFS custom resource definition. For further information please refer to the example [here](https://rook.io/docs/rook/latest/CRDs/ceph-nfs-crd/#example).
+
```

Comment:
There are several missing CRDs, could we add all these with this PR? Here is the full list:
```
$ k create -f crds.yaml 
customresourcedefinition.apiextensions.k8s.io/cephblockpoolradosnamespaces.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephblockpools.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephbucketnotifications.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephbuckettopics.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephclients.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephclusters.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephfilesystemmirrors.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephfilesystems.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephfilesystemsubvolumegroups.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephnfses.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephobjectrealms.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephobjectstores.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephobjectstoreusers.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephobjectzonegroups.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephobjectzones.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/cephrbdmirrors.ceph.rook.io created
customresourcedefinition.apiextensions.k8s.io/objectbucketclaims.objectbucket.io created
customresourcedefinition.apiextensions.k8s.io/objectbuckets.objectbucket.io created
```